core/src/main/java/com/linecorp/armeria/server/VirtualHost.java

```diff
@@ -208,6 +235,21 @@ void setServerConfig(ServerConfig serverConfig) {
         router.registerMetrics(registry, idPrefix);
     }
 
+    /**
+     * Sets the {@link Logger} which is used for writing access logs of this virtual host.
+     */
+    void accessLogger(Logger logger) {
+        accessLogger = requireNonNull(logger, "logger");
+    }
+
+    /**
+     * Returns the {@link Logger} which is used for writing access logs of this virtual host.
+     */
+    @Nullable
+    public Logger accessLogger() {
```

Comment:
How about removing `@Nullable` here? e.g.

```java
public Logger accessLogger() {
    checkState(accessLogger != null, "accessLogger not initialized yet");
    return accessLogger;
}

@Nullable 
Logger accessLoggerOrNull() { // Use this method internally
    return accessLogger;
}
```

.. because a user should not worry about nullness of `accessLogger`.